src/main/java/net/sf/jabref/remote/DBProcessor.java

```diff
@@ -0,0 +1,551 @@
+/*  Copyright (C) 2003-2016 JabRef contributors.
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+package net.sf.jabref.remote;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import net.sf.jabref.MetaData;
+import net.sf.jabref.event.location.EntryEventLocation;
+import net.sf.jabref.exporter.FieldFormatterCleanups;
+import net.sf.jabref.logic.cleanup.FieldFormatterCleanup;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.model.entry.BibEntry;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * Processes all incoming or outgoing bib data to external SQL Database and manages its structure.
+ */
+public class DBProcessor {
+
+    private static final Log LOGGER = LogFactory.getLog(DBConnector.class);
+
+    private Connection connection;
+
+    private DBType dbType;
+    private final DBHelper dbHelper;
+
+    public static final String ENTRY = "ENTRY";
+    public static final String METADATA = "METADATA";
+
+    public static final List<String> ALL_TABLES = new ArrayList<>(Arrays.asList(ENTRY, METADATA));
+
+    // Elected column names of main the table
+    // This entries are needed to ease the changeability, cause some database systems dependent on the context expect low or uppercase characters.
+    public static final String ENTRY_REMOTE_ID = "REMOTE_ID";
+    public static final String ENTRY_ENTRYTYPE = "ENTRYTYPE";
+
+    public static final String METADATA_SORT_ID = "SORT_ID";
+    public static final String METADATA_KEY = "META_KEY";
+    public static final String METADATA_FIELD = "FIELD";
+    public static final String METADATA_VALUE = "META_VALUE";
+
+
+    /**
+     * @param connection Working SQL connection
+     * @param dbType Instance of {@link DBType}
+     */
+    public DBProcessor(Connection connection, DBType dbType) {
```

Comment:
In my opinion the `DBProcessor` shouldn't directly depend on the connection but only on the wrapper `DBHelper`. In this way you can mock `DBHelper` in your tests and check that the correct queries are generated.